### PR TITLE
Bugfix: dont create empty ack files

### PR DESCRIPTION
**Pull request by spuun**
### WHAT is this pull request doing?
Fixes a bug where ack files for each segment was created during startup.

### HOW can this pull request be tested?
Run specs

### CODE REVIEW

**Comment by spuun** [comment]
recheck

**Comment by spuun** [comment]
I have read the CLA Document and I hereby sign the CLA

**Review by kickster97 — APPROVED** [review]

File: src/lavinmq/queue/message_store.cr [hunk 1]
```diff
@@ -359,7 +359,7 @@ module LavinMQ
         @segments.reject! do |seg, mfile|
           next if seg == current_seg # don't the delete the segment still being written to
 
-          if @segment_msg_count[seg] == @acks[seg].size // sizeof(UInt32)
+          if (acks = @acks[seg]?) && @segment_msg_count[seg] == (acks.size // sizeof(UInt32))
```
Comment (carlhoerberg) [inline src/lavinmq/queue/message_store.cr:362]: Just tried to make it more readable, don't know if i succeeded... 

```suggestion
          if @acks.has_key?(seg) && @segment_msg_count[seg] == @acks[seg].size // sizeof(UInt32)
```

```suggestion
          if @segment_msg_count[seg] == (@acks[seg]?.try(&.size) || 0) // sizeof(UInt32)
```

Comment (spuun) [reply src/lavinmq/queue/message_store.cr:362]: I think I did the way I did to get rid of one access to the instance variable. Imo the `has_key?` check is the most readable, but maybe
```crystal
if @segment_msg_count[seg] == (@acks.fetch(seg, 0) // sizeof(UInt32))
```
is as good. `try` and `||` isn't very readable imo.

Comment (carlhoerberg) [reply src/lavinmq/queue/message_store.cr:362]: Yes, even better, thanks!

**Pull request merged**